client/cypress/integration/visualizations/counter_spec.js

```diff
@@ -0,0 +1,207 @@
+  SELECT 31415.9265359 AS a, 40000 AS b, 'ipsum' AS c
+`;
+
+describe('Counter', () => {
+  const viewportWidth = Cypress.config('viewportWidth');
+
+  beforeEach(() => {
+    cy.login();
+    createQuery({ query: SQL }).then(({ id }) => {
+      cy.visit(`queries/${id}/source`);
+      cy.getByTestId('ExecuteButton').click();
+    });
+  });
+
+  it('creates simple Counter', () => {
+    cy.clickThrough(`
+      NewVisualization
+      VisualizationType
+      VisualizationType.COUNTER
+
+      Counter.General.ValueColumn
+      Counter.General.ValueColumn.a
+    `);
+
+    cy.getByTestId('VisualizationPreview').find('.counter-visualization-container').should('exist');
+
+    // wait a bit before taking snapshot
+    cy.wait(500); // eslint-disable-line cypress/no-unnecessary-waiting
+    cy.percySnapshot('Visualizations - Counter (format target value)', { widths: [viewportWidth] });
+  });
+
+  it('creates Counter with custom label', () => {
+    cy.clickThrough(`
+      NewVisualization
+      VisualizationType
+      VisualizationType.COUNTER
+
+      Counter.General.ValueColumn
+      Counter.General.ValueColumn.a
+    `);
+
+    cy.fillInputs({
+      'Counter.General.Label': 'Custom Label',
+    });
+
+    cy.getByTestId('VisualizationPreview').find('.counter-visualization-container').should('exist');
+
+    // wait a bit before taking snapshot
+    cy.wait(500); // eslint-disable-line cypress/no-unnecessary-waiting
+    cy.percySnapshot('Visualizations - Counter (format target value)', { widths: [viewportWidth] });
+  });
+
+  it('creates Counter with non-numeric value', () => {
+    cy.clickThrough(`
+      NewVisualization
+      VisualizationType
+      VisualizationType.COUNTER
+
+      Counter.General.ValueColumn
+      Counter.General.ValueColumn.c
+
+      Counter.General.TargetValueColumn
+      Counter.General.TargetValueColumn.c
+    `);
+
+    cy.fillInputs({
+      'Counter.General.TargetValueRowNumber': '2',
+    });
+
+    cy.getByTestId('VisualizationPreview').find('.counter-visualization-container').should('exist');
+
+    // wait a bit before taking snapshot
+    cy.wait(500); // eslint-disable-line cypress/no-unnecessary-waiting
+    cy.percySnapshot('Visualizations - Counter (format target value)', { widths: [viewportWidth] });
+  });
+
+  it('creates Counter with target value (trend positive)', () => {
+    cy.clickThrough(`
+      NewVisualization
+      VisualizationType
+      VisualizationType.COUNTER
+
+      Counter.General.ValueColumn
+      Counter.General.ValueColumn.a
+
+      Counter.General.TargetValueColumn
+      Counter.General.TargetValueColumn.b
+    `);
+
+    cy.getByTestId('VisualizationPreview').find('.counter-visualization-container').should('exist');
+
+    // wait a bit before taking snapshot
+    cy.wait(500); // eslint-disable-line cypress/no-unnecessary-waiting
+    cy.percySnapshot('Visualizations - Counter (format target value)', { widths: [viewportWidth] });
```

Comment:
It seems all the Percy Snapshots have the same name :sweat_smile:, guess it will mess with Percy...

Reply:
Uh-oh... Thanks! Probably a copy-paste leftover...

Reply:
Fixed :+1: